### PR TITLE
[Inditex]bug-1937618, retain bucket policies at archive site

### PR DESCRIPTION
**Pull request by viduship**
Automate bug-1937618, Retain bucket policies at the archive site

Bucket policies disappears in archive zone when an object is inserted in master zone bucket

https://bugzilla.redhat.com/show_bug.cgi?id=1937618

https://polarion.engineering.redhat.com/polarion/#/project/https://polarion.engineering.redhat.com/polarion/redirect/project/CEPH/workitem?id=CEPH-83575576

logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-D764QV

### CODE REVIEW

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 1]
```diff
@@ -239,6 +240,15 @@ def test_exec(config, ssh_con):
...
                 )
+                if config.retain_bucket_pol:
+                    log.info(
+                        "Test bucket policy retained at archive site after writing IOs bug-1937618"
+                    )
+                    reusable.retain_bucket_policy(
+                        rgw_conn2, bucket_name_to_create, config
+                    )
+
+                bucket_name_to_create
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:251]: Looks like this is an unwanted line..please remove it.

Comment (viduship) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py]: done.

**Review by ckulal — COMMENTED** [review]
Looks good

File: rgw/v2/tests/s3_swift/reusable.py [hunk 2]
```diff
@@ -85,6 +85,19 @@ def set_get_object_acl(
...
 
+def retain_bucket_policy(rgw_conn2, bucket_name_to_create, config):
+    """
+    put bucket policy and retain it at archive site
+    """
+    log.info(f"Set bucket policy on {bucket_name_to_create}")
+    put_bucket_pol = rgw_conn2.put_bucket_policy(
+        Bucket=bucket_name_to_create,
+        Policy='{"Version": "2012-10-17", "Statement": [{ "Sid": "id-1","Effect": "Allow","Principal": {"AWS": "arn:aws:iam::123456789012:root"}, "Action": [ "s3:PutObject","s3:PutObjectAcl"], "Resource": ["arn:aws:s3:::acl3/*" ] } ]}',
+    )
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/reusable.py:96]: Can we have policy as one of the argument to this function??
so that we can provide it dynamically

Comment (viduship) [reply rgw/v2/tests/s3_swift/reusable.py:96]: @anrao19 , this is just a random bucket policy just for testing if it is retained after any IOs, but we are actually not testing the policy here.

**Review by anrao19 — APPROVED** [review]

**Review by hmaheswa — APPROVED** [review]

**Review by ckulal — APPROVED** [review]

**Pull request merged**